### ISSUE TITLE
Editable app-model keybindings at runtime

### ISSUE THREAD

**Issue by lucyleeow** [napari/napari#6600]
First discussed here: https://github.com/napari/napari/issues/6516#issuecomment-1886127528

Currently actions are being migrated to app-model, from action manager (see: https://github.com/napari/napari/issues/4860 and https://github.com/napari/napari/pull/6204, which migrates layer and viewer actions). Keybindings are defined in app-model Actions, and registered in the app-model keybinding registry when the action is registered.

It is  intended that app-model keybindings will be editable at runtime and the plan is for them to be able to be edited by the user.
Currently you can update the app-model keybinding registry at runtime BUT this will not over-ride an existing keybinding (will require a PR in app-model to update this).

There are 3 levels of keybinding setting, from lowest to highest priority:

1. Native napari app
2. Plugins will be able to set keybindings
   * Implemented in https://github.com/napari/napari/pull/6204
3. User can set keybindings

The app-model `app` keybinding registry should store 'source' information (similar to VS code). The app will use the keybinding from the highest priority source. We will know when a keybinding comes from native napari or plugin (via manifest). Any setting of keybinding at runtime will be user set. The user should be able to do this via GUI or a new API function in headless mode (as we do not want to user to have to set 'source'). This addition will require a PR in app-model, to enable the app-model `app` to store 'source' info. When creating the app/keybinding registry, we should be able to set the possible 'source's and what priority they will be.

Note that keybinding rules also has a weight parameter (you could have >1 keybinding from the same source with different weights) but we don't think we want to expose this and would go with YAGNI.

Related but parallel: https://github.com/napari/napari/pull/6204 implements nap 7. This should probably implemented after #6204.

**Comment by lucyleeow** [comment]
cc @jni @aganders3 @DragaDoncila

**Comment by DragaDoncila** [comment]
Looking a bit more closely at #6204 it looks like what we'd be doing is abstracting some of the [implementation](https://github.com/napari/napari/pull/6204/files#diff-f79e526e98f9ab2565d94695b7d890c6b0c39f32a21b183ce38a34b185e85b0c) of the `QKeyBindingDispatcher` to `app_model` as a `Qt` backend (frontend?) for the keybinding registry.

**Comment by DragaDoncila** [comment]
Afaik #6204 implements the editable aspect as well? Also we may want to expose menu shortcuts as well, not sure

**Comment by lucyleeow** [comment]
Ooohhhhh it does? I have not looked deeply. Let's take this offline

**Comment by lucyleeow** [comment]
closing this in favour of #7059
